### DISCUSSION TITLE
Full orchestra defn-spec support for lsp usage.

### DISCUSSION THREAD

**Discussion by ghost** [clj-kondo/clj-kondo#1994] [Q&A]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
def-catch-all works for linting. If I want to use find-all-references in lsp, I've got to type the function name all by itself and invoke it there. I can not invoke it within defn-spec itself. There is something almost correct, autocomplete and the little tag in sider that says "2 references" do not work, but the syntax highlighting is fine. 

**Describe the solution you'd like**
I've seen it discussed in the archives of slack, but I haven't seen anyone come up with an answer around auto-complete and goto reference for lsp. There may already be a solution and I just was using the 

**Describe alternatives you've considered**
Borkdude said,
"clojure-lsp should at least "see" that defn-spec defines a var, since this is what clj-kondo is providing in the analysis with this config. And clojure-lsp uses the analysis from clj-kondo."

in this reddit thread,
https://www.reddit.com/r/Clojure/comments/r66ygy/using_orchestra_with_clojurelsp/

Since I'm seeing the same issue, I imagine there's something lost in translation, but I'm at a loss for ways to go about remediation on my own.

**Comment by borkdude** [comment]
@netdrilling Can you make a repro repository that I can clone locally to test your problem?

**Reply by ghost** [reply]
You're quick on the jump. Much love for taking a look. You're great. There's a screenshot in the project root that will show the behavior on my end. 

[repro repo](https://github.com/netdrilling/defn-spec-kondo-troubleshooting)

**Reply by borkdude** [reply]
Interesting, I can reproduce the problem. Looking closer.

**Reply by borkdude** [reply]
There was indeed a problem with the locations that were emitted. This should now work on clj-kondo master. See issue #1995

**Reply by borkdude** [reply]
![defn-spec](https://user-images.githubusercontent.com/284934/219141759-817ee65a-bcbf-4c4a-9a6e-3ca630644082.gif)

**Reply by ghost** [reply]
Just wanted to check in with my thanks. I saw that Eric bumped the version of clj-kondo today, so I finally upgraded. CHEERS! I'm so happy. My macros and defn-spec kicking with the definition jump around, docs on hover, all that. Great work between the two of yall. Much love!

**Reply by borkdude** [reply]
@netdrilling Thanks for re-confirming!
